proxy/proxy.go

```diff
@@ -490,7 +498,9 @@ func mapRequest(ctx *context, requestContext stdlibcontext.Context, removeHopHea
 		setRequestURLFromRequest(u, r)
 		setRequestURLForDynamicBackend(u, stateBag)
 	case eskip.LBBackend:
-		endpoint = setRequestURLForLoadBalancedBackend(u, rt, &routing.LBContext{Request: r, Route: rt, LBEndpoints: rt.LBEndpoints, Params: stateBag, Registry: registry})
+		endpoint = selectEndpoint(ctx, registry)
+		u.Scheme = endpoint.Scheme
```

Comment:
minor: maybe use fallthrough here?

Reply:
no need to overcomplicate to save 1 line

Reply:
also it uses endpoint fields, not route fields